Modules/socketmodule.c

```diff
@@ -758,6 +758,8 @@ internal_select(PySocketSockObject *s, int writing, _PyTime_t interval,
 
     Py_BEGIN_ALLOW_THREADS;
     n = poll(&pollfd, 1, (int)ms);
+    if (pollfd.revents & (POLLERR | POLLNVAL))
+        n = -1;
```

Comment:
The caller of `internal_select` probably expects `errno` to be set if `internal_select` returns `-1`. For `POLLINVAL`, I suppose we can just set `errno` to `EBADF`. For `POLLERR`, the best we can do may be run the outer operation and hope it returns the underlying error.

Does `POLLHUP` also need some handling here?